### ISSUE TITLE
hcloud-csi-node CrashLoopBackOff v2.1.0

### ISSUE THREAD

**Issue by kiwinesian** [hetznercloud/csi-driver#347]
hi there,

so,, just going back to the Hetzner CSI but couldn't get it to work for some reason.
Configuration:
1. K8S 1.24.8
2. CSI v2.1.0
3. Also running latest Hetzner CCM
4. Cilium 1.12 as the CNI

After few hours, I kept running to `Still connecting to unix:///run/csi/socket` on the `hcloud-csi-node`
![image](https://user-images.githubusercontent.com/12476195/208615249-b4a609a4-0037-489a-9b1f-52078831f9ca.png)

`hcloud-csi-controller` looks to be running fine, all the containers are healthy
![image](https://user-images.githubusercontent.com/12476195/208615379-cbad5a51-cee9-434a-a53e-fdb4f871e98b.png)
`hetznercloud/hcloud-csi-driver:2.1.0` also seems to be running on the controller side

however, on each of the `hcloud-csi-node`, this is where things are failing as below,
![image](https://user-images.githubusercontent.com/12476195/208615633-4adc687f-f2e2-403d-83ed-299a1d8f926d.png)
![image](https://user-images.githubusercontent.com/12476195/208615790-71b0c40a-13ea-46da-ad33-e05f95fa1bc2.png)
![image](https://user-images.githubusercontent.com/12476195/208615842-93cfbb79-da0d-41cd-ab15-3884d98e1d85.png)
![image](https://user-images.githubusercontent.com/12476195/208615897-a197778d-c226-4694-9bce-7256671b6b40.png)

What could've gone wrong? Cilium config? Firewall? driver issue?
Just a little confused given that the `controller` went all well, but not the rest of the `csi-node`

below is the log file from the `controller`
![image](https://user-images.githubusercontent.com/12476195/208616259-88cef004-5b1e-4103-9aca-f5f799d4a18c.png)

thanks so much! 🙏

**Comment by apricote** [comment]
Hey @kiwinesian,

can you [enable the debug logs](https://github.com/hetznercloud/csi-driver/blob/main/docs/debug-logs.md) and then send me the logs for the new pods from following command?

```
kubectl logs -n kube-system -lapp=hcloud-csi -c hcloud-csi-driver --previous --prefix
```

You can either post the logs here, or send them through email to `julian.toelle <at> hetzner-cloud.de`

**Comment by kiwinesian** [comment]
hi @apricote ,

thanks for reaching back - much appreciated!

See below for the response

```
> kubectl logs -n kube-system -lapp=hcloud-csi -c hcloud-csi-driver --previous --prefix
[pod/hcloud-csi-node-2k44b/hcloud-csi-driver] level=debug ts=2022-12-20T17:13:53.18348602Z component=metrics-service msg="registering metrics with registry"
[pod/hcloud-csi-node-2k44b/hcloud-csi-driver] level=debug ts=2022-12-20T17:13:53.183658586Z component=metrics-service msg="registered metrics"
[pod/hcloud-csi-node-2k44b/hcloud-csi-driver] level=debug ts=2022-12-20T17:13:53.183678606Z component=metrics-service msg="starting prometheus http server" addr=0.0.0.0:9189
[pod/hcloud-csi-node-7xx66/hcloud-csi-driver] level=debug ts=2022-12-20T17:13:53.627842365Z component=metrics-service msg="registering metrics with registry"
[pod/hcloud-csi-node-7xx66/hcloud-csi-driver] level=debug ts=2022-12-20T17:13:53.62795895Z component=metrics-service msg="registered metrics"
[pod/hcloud-csi-node-7xx66/hcloud-csi-driver] level=debug ts=2022-12-20T17:13:53.627984539Z component=metrics-service msg="starting prometheus http server" addr=0.0.0.0:9189
[pod/hcloud-csi-node-gww8z/hcloud-csi-driver] level=debug ts=2022-12-20T17:13:55.569287242Z component=metrics-service msg="registering metrics with registry"
[pod/hcloud-csi-node-gww8z/hcloud-csi-driver] level=debug ts=2022-12-20T17:13:55.569456777Z component=metrics-service msg="registered metrics"
[pod/hcloud-csi-node-gww8z/hcloud-csi-driver] level=debug ts=2022-12-20T17:13:55.569473201Z component=metrics-service msg="starting prometheus http server" addr=0.0.0.0:9189
[pod/hcloud-csi-node-tkb64/hcloud-csi-driver] level=debug ts=2022-12-20T17:13:52.139993428Z component=metrics-service msg="registering metrics with registry"
[pod/hcloud-csi-node-tkb64/hcloud-csi-driver] level=debug ts=2022-12-20T17:13:52.140204165Z component=metrics-service msg="registered metrics"
[pod/hcloud-csi-node-tkb64/hcloud-csi-driver] level=debug ts=2022-12-20T17:13:52.140228493Z component=metrics-service msg="starting prometheus http server" addr=0.0.0.0:9189
```

I can send it over to the provided email address if needed! :)

**Comment by samcday** [comment]
Hiya @kiwinesian. Could you let us know if you upgraded your csi-driver deployment from a previous version? If so, do you know from which version you upgraded?

Could you also please provide the output of `kubectl get ds/hcloud-csi-node -o yaml`?

**Comment by samcday** [comment]
Oh and also, could you please let us know what container runtime you're using?

**Comment by kiwinesian** [comment]
heyyy @samcday,

thanks for reaching back!
spent the last 2 days to look into this and played around with different build.
Looks like the issue was due to Cilium configuration and not necessarily a bug!

I found a configuration that will also work with the Hetzner CSI - so may not be a bug at all.

more than happy to continue with the investigation if you think that it is a bug - just let me know what needs to be done from my cluster! :)

is running the latest version containerd://1.6.14, kernel 5.15.0-56-generic, and Ubuntu 22.04.1 LTS
